### PR TITLE
Add zsh completion to svcat

### PR DESCRIPTION
**Pull request by kikisdeliveryservice**
closes: #1912 

This is my first PR for svcat.

Issue 1912 requires two steps to close: 

- [x] 1. bump spf13/cobra to 0.0.1 (or above) 
- [x] 2. add zsh completion - zsh completion for svcat is added

### CODE REVIEW

File: cmd/svcat/completion/completion_cmd.go [hunk 1]
```diff
@@ -117,3 +119,154 @@ func (c *completionCmd) Run() error {
 func runCompletionBash(w io.Writer, cmd *cobra.Command) error {
 	return cmd.Root().GenBashCompletion(w)
 }
+
+func runCompletionZsh(out io.Writer, cmd *cobra.Command) error {
```
Comment (carolynvs) [inline cmd/svcat/completion/completion_cmd.go]: Can we replace this function's body with this instead?

```golang
    return cmd.Root().GenZshCompletion(w)
```

**Comment by kikisdeliveryservice** [comment]
Hi @carolynvs  I wasn't able to get the GenZshCompletion() working a few iterations back so I ended up implementing the kubectl / helm way (which does indeed work).  

As requested I re-implemented using cobra's GenZshCompletion.  It doesn't seem to be working for me still but you can pull and try it. We can always revert back to the previous change if you confirm using GenZshCompletion() is no good.

File: cmd/svcat/completion/completion_cmd.go (hunk 1 above)
Comment (carolynvs) [reply cmd/svcat/completion/completion_cmd.go]: Turns out that the workaround you have is the best way at the moment, cobra is still working on getting the zsh functionality 100%. 👍

**Comment by kikisdeliveryservice** [comment]
@carolynvs 
1. yes!
2. svcat get b<TAB> actually has two options: bindings and brokers. so can confirm:
svcat get b<TAB> --> shows on next line:  `bindings  brokers`
svcat get bi<TAB> --> `svcat get bindings`
svcat get br<TAB> --> `svcat get brokers`

**Comment by arschles** [comment]
LGTM, but I'll let folks from another org give the final LGTM label

File: cmd/svcat/completion/completion_cmd.go (hunk 1 above)
Comment (n3wscott) [reply cmd/svcat/completion/completion_cmd.go]: I would recommend moving the entire function to it's own file, like `completion_cmd_zsh.go`

Comment (n3wscott) [reply cmd/svcat/completion/completion_cmd.go]: well, meh... the file only doubles and it is ~250 likes. I take it back.

**Pull request merged**